WordPress/Classes/ViewRelated/Stats/Insights/ViewsVisitors/ViewsVisitorsLineChartCell.swift

```diff
@@ -54,14 +54,34 @@ struct StatsSegmentedControlData {
     }
 
     var differenceLabel: String {
-        let stringFormatValue = differencePercent != 0 ? "%@%@ (%@%%)" : "%@%@"
-        let stringFormat = NSLocalizedString(stringFormatValue, comment: "Difference label for Insights Overview stat, indicating change from previous period. Ex: +99.9K(5%)")
-        return String.localizedStringWithFormat(
-            stringFormat,
-            difference < 0 ? "" : "+",
-            difference.abbreviatedString(),
-            differencePercent.abbreviatedString()
-        )
+        // We want to show something like "+1.2K (5%)" if we have a percentage difference and "1.2K" if we don't.
+        // Because localized strings need to be strings literal, we cannot embed any conditional logic in the `localizedString...` call.
+        // We therefore need to generate different string literals base on the state.
+        let differenceSign = difference < 0 ? "" : "+"
+        if differencePercent != 0 {
+            let stringFormat = NSLocalizedString(
+                "insights.visitorsLineChartCell.differenceLabelWithPercentage",
+                value: "%@%@ (%@%%)",
+                comment: "Difference label for Insights Overview stat, indicating change from previous period, including percentage value. Example: +99.9K (5%)"
+            )
+            return String.localizedStringWithFormat(
+                stringFormat,
+                differenceSign,
+                difference.abbreviatedString(),
+                differencePercent.abbreviatedString()
+            )
+        } else {
+            let stringFormat = NSLocalizedString(
+                "insights.visitorsLineChartCell.differenceLabelWithoutPercentage",
+                value: "%@%@",
+                comment: "Difference label for Insights Overview stat, indicating change from previous period. Example: +99.9K"
+            )
+            return String.localizedStringWithFormat(
+                stringFormat,
+                differenceSign,
+                difference.abbreviatedString()
+            )
+        }
```

Comment:
While I was at it, I also added reverse-DNS keys to that we can avoid ambiguous translations. See #19028.

FYI @staskus, as per [this commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/19400/commits/8545f1cce030165e5eb88635fcbb063a40bc9d05).

Also, I noticed there is no unit test for this logic. It would be great to add some, even though they wouldn't have caught this issue in particular.

Reply:
@mokagio 

Thank you for noticing the issue and making the fixes! 🤝

Should I add additional unit tests and additional cases that @AliSoftware was mentioning as a part of this PR?